### PR TITLE
Fixes #39: Create vercel deployment workflow

### LINKED ISSUE #39: Create vercel deployment workflow
> There are two options I can think of if we don't want to do that,

https://vercel.com/docs/cli/deploy#deploying-to-a-custom-domain

> In the following example, you create a bash script that you include in your CI/CD  workflow. The goal is to have all preview deployments be aliased to a custom domain so that developers can bookmark the preview deployment URL.

if I read it correctly, you can get vercel deploy to auto-generate a url that you can then alias to the right environment.

    vc alias $deploymentUrl finesse-frontend-dev.vercel.com

or

    vc alias $deploymentUrl finesse-frontend-uat.vercel.com

or 

    vc alias $deploymentUrl finesse-frontend-prd.vercel.com

_Originally posted by @rngadam in https://github.com/ai-cfia/finesse-frontend/issues/38#issuecomment-1828342204_
 
 Creation of 3 workflow for each development stage need to be added

### PR DESCRIPTION
**Pull request by redat97**
#39

### CODE REVIEW

File: .github/workflows/workflow-vercel-deployment.yml [hunk 1]
```diff
@@ -1,11 +1,22 @@
 name: Vercel Preview Deployment
 
+on:
+  workflow_call:
+    inputs:
+      vercel-project-name:
```
Comment (rngadam) [inline .github/workflows/workflow-vercel-deployment.yml:6]: as discussed in the ADR, we shouldn't exposed implementation specific naming. 

could we just call this project-name as a parameter?

File: .github/workflows/workflow-vercel-deployment.yml [hunk 2]
```diff
@@ -1,11 +1,22 @@
 name: Vercel Preview Deployment
```
Comment (rngadam) [inline .github/workflows/workflow-vercel-deployment.yml:1]: Vercel Static Frontend Deployment

File: .github/workflows/workflow-vercel-deployment.yml [hunk 3]
```diff
@@ -1,11 +1,22 @@
...
 
+on:
+  workflow_call:
+    inputs:
+      vercel-project-name:
+        required: true
+        type: string
+      deployment_environment:
```
Comment (rngadam) [inline .github/workflows/workflow-vercel-deployment.yml:9]: using underscore here but dashes in the other input parameter

naming conventions for the workflow should be added to our naming ADR

File: .github/workflows/workflow-vercel-deployment.yml [hunk 4]
```diff
@@ -16,18 +27,14 @@ jobs:
...
       - name: Install Vercel CLI
         run: npm install --global vercel@latest
       - name: Pull Vercel Environment Information
         run: vercel pull --yes --environment=preview --token=${{ secrets.VERCEL_TOKEN }}
       - name: Build Project Artifacts
         run: vercel build --token=${{ secrets.VERCEL_TOKEN }}
       - name: Deploy Project Artifacts to Vercel
-        run: vercel deploy --prebuilt --token=${{ secrets.VERCEL_TOKEN }}
-        
-      - name: Comment on PR
```
Comment (rngadam) [inline .github/workflows/workflow-vercel-deployment.yml:29]: why was the comment on PR removed? Was it already displayed?

**Review by rngadam — CHANGES_REQUESTED** [review]

File: .github/workflows/workflow-vercel-deployment.yml (hunk 4 above)
Comment (redat97) [reply .github/workflows/workflow-vercel-deployment.yml:29]: I found it useless because it was a set url that feature was implemented to replace vercel's bot which comment the preview URL (not static or whatever the term is not sure) but I'll put it back if you liked it

Comment (rngadam) [reply .github/workflows/workflow-vercel-deployment.yml:29]: can you show me a screenshot or link to an example before/after this change? how do we find out which URL it is deployed to now?

Comment (redat97) [reply .github/workflows/workflow-vercel-deployment.yml:29]: it's from the logs there's not really a before after I just figured since the everyone knows the URL and the bot tend to get annoying commenting each time you push to the branch it creates a bunch of comments and it can make it hard to follow discussions I will try to implement it back but only on PR creation

File: .github/workflows/workflow-vercel-deployment.yml (hunk 1 above)
Comment (redat97) [reply .github/workflows/workflow-vercel-deployment.yml]: yeah sounds good I changed it

File: .github/workflows/workflow-vercel-deployment.yml (hunk 2 above)
Comment (redat97) [reply .github/workflows/workflow-vercel-deployment.yml]: done

File: .github/workflows/workflow-vercel-deployment.yml (hunk 4 above)
Comment (redat97) [reply .github/workflows/workflow-vercel-deployment.yml:29]: I added it back

**Review by rngadam — APPROVED** [review]
